### PR TITLE
Fix: Support LPS in psp6 format

### PR DESCRIPTION
**Pull request by sunliang98**
### Reminder
- [ ] Have you linked an issue with this pull request?
- [ ] Have you noticed possible changes of behavior below or in the linked issue?
- [ ] Have you explained the changes of codes in core modules of ESolver, HSolver, ElecState, Hamilt, Operator or Psi? (ignore if not applicable)

### Linked Issue
Fix #...

### What's changed?
Support Local pseudopotential in psp6 format.
Add an integrate test 101_PW_lps6_pseudopotential.

### Any changes of core modules? (ignore if not applicable)
- Example: I have added a new virtual function in the esolver base class in order to ...

### CODE REVIEW

**Comment by kirk0830** [comment]
Amazing, but I am curious about the construction of this pseudopotential, is it usually for norm-conserving, or ultrasoft? I see you decrease the ecutwfc in one intergrated test.

File: source/module_cell/read_pp_blps.cpp
Comment (kirk0830) [inline source/module_cell/read_pp_blps.cpp:1]: I have read manually the ag_lps.cpi file you upload for test and find it is organized like a plain-text, other than json or xml (how upf usually organized). I am a little bit concern about the range of compatibility of the way to parse pseudopotential here. According to your knowledge, is there any exception that the function `read_pseudo_blps` cannot parse properly?

**Review by kirk0830 — COMMENTED** [review]
A new pseudopotential format parse is supported, but given upf is common in open-source pseudopotentials, and psp6 or pso8 are relatively scarsely seen, could you leave a breif intro. about what softwares use this kinds of pseudopotential?

**Comment by sunliang98** [comment]
The psp6 and psp8 format are usually used by the norm-conserving pseudopotentials utilized by ABINIT, one can find the introduction of file structure in https://docs.abinit.org/developers/psp6_info/ and https://docs.abinit.org/developers/psp8_info/. Besides, some local pseudopotentials (usually used in OFDFT) are organized in the psp6 and psp8 format, so that we support them here. we note that `read_pseudo_blps` ONLY support local pseudopotentials in psp6 and psp8 format, NOT norm-conserving pseudopotentials.

**Comment by sunliang98** [comment]
> Amazing, but I am curious about the construction of this pseudopotential, is it usually for norm-conserving, or ultrasoft? I see you decrease the ecutwfc in one intergrated test.

The ecutwfc is decreased only to accelerate the integrate test.

**Review by kirk0830 — APPROVED** [review]

**Review by Qianruipku — APPROVED** [review]

**Pull request merged**